src/js/components/awardv2/idv/activity/ActivityChartTooltip.jsx

```diff
@@ -78,48 +80,94 @@ export default class IdvActivityTooltip extends React.Component {
             this.positionTooltip();
         });
     }
+    // uses a ratio of
+    // someDiv / GraphWidth = x ( TruncatedWidth ) / shortenedTooltipWidth
+    // to get what the shortened width of the div should be
+    truncatedWidth(graphWidth, tooltipWidth, div) {
+        return (graphWidth * div) / tooltipWidth;
+    }
+    // uses a ratio of
+    // TruncatedDivWidth / NormalDivWidth =
+    // x ( Truncated character length ) / NormalCharacterLength
+    truncatedCharacterLength(truncatedDivWidth, normalDivWidth, normalCharacterLength) {
+        return Math.floor(
+            ((truncatedDivWidth * normalCharacterLength) / normalDivWidth) / 2);
+    }
+    // truncates the text if warranted
+    truncateText(text, truncatedLength, propertyName) {
+        if (truncatedLength < text.length) {
+            const truncatedText = `${text
+                .substring(0, truncatedLength).trim()}...`;
+            return { [propertyName]: truncatedText };
+        }
+        return { [propertyName]: text };
+    }
 
-    truncateText(awardingAgencyNameDiv, recipientNameDiv, theTooltipWidth) {
+    truncateLogic(arrayOfDivsToTruncate, theTooltipWidth) {
```

Comment:
This name of this first parameter `arrayOfDivsToTruncate` seems to indicate that the value of this array is actual refs to divs.

Could we rename this to reflect the real value, which is integers reflecting an the width of each div (?).

Perahps `arrayOfDivWidths`?